clients/urlclient/interfaces/retry.go

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// interfaces defines contracts specific to the urlclient package
+package interfaces
+
+// RetryStrategy defines some way to verify that a RegistryClient has received data.
+// It accomplishes this by monitoring the values pointed to in its parameters,
+// which communicate the state of the RegistryClient.
+// When the values pointed to by the parameters are in the desired state,
+// the URL that should be passed back to the RegistryClient should be returned by the RetryStrategy.
+type RetryStrategy interface {
+	// Retry defines the actual behavior of the RegistryClient and how it processes the URL.
+	Retry(url *string) (string, error)
```

Comment:
Can we change the name from `url` to something more general like `value`. I only suggest that because some of the comments use the word `poll` and that mixed with `url` can possibly lead to believe that the `RetryStrategy` is polling that URL. I made this mistake when I jumped right into the code at first. If this is silly then ignore this comment.